cmd/broker/provisioning_test.go

```diff
@@ -2088,3 +2088,132 @@ func TestProvisioning_BuildRuntimePlans(t *testing.T) {
 		suite.AssertRuntimeResourceLabels(opID)
 	})
 }
+
+func TestProvisioning_ResolveSubscriptionSecretStepEnabled(t *testing.T) {
+	for tn, tc := range map[string]struct {
+		planID         string
+		region         string
+		platformRegion string
+
+		expectedProvider         string
+		expectedSubscriptionName string
+	}{
+		"Trial": {
+			planID: broker.TrialPlanID,
+
+			expectedProvider:         "aws",
+			expectedSubscriptionName: "sb-aws-shared",
+		},
+		"Freemium aws": {
+			planID:         broker.FreemiumPlanID,
+			region:         "eu-central-1",
+			platformRegion: "cf-eu10",
+
+			expectedProvider:         "aws",
+			expectedSubscriptionName: "sb-aws",
+		},
+		"Freemium azure": {
+			planID:         broker.FreemiumPlanID,
+			region:         "westeurope",
+			platformRegion: "cf-eu21",
+
+			expectedProvider:         "azure",
+			expectedSubscriptionName: "sb-azure",
+		},
+		"Production Azure": {
+			planID: broker.AzurePlanID,
+			region: "westeurope",
+
+			expectedProvider:         "azure",
+			expectedSubscriptionName: "sb-azure",
+		},
+		"Production AWS": {
+			planID: broker.AWSPlanID,
+			region: "us-east-1",
+
+			expectedProvider:         "aws",
+			expectedSubscriptionName: "sb-aws",
+		},
+		"Production GCP": {
+			planID: broker.GCPPlanID,
+			region: "us-central1",
+
+			expectedProvider:         "gcp",
+			expectedSubscriptionName: "sb-gcp",
+		},
+		"Production GCP KSA": {
+			planID:         broker.GCPPlanID,
+			region:         "me-central2",
+			platformRegion: "cf-sa30",
+
+			expectedProvider:         "gcp",
+			expectedSubscriptionName: "sb-gcp_cf-sa30",
+		},
+		"sap converged cloud eu-de-1": {
+			planID:         broker.SapConvergedCloudPlanID,
+			region:         "eu-de-1",
+			platformRegion: "cf-eu20-staging",
+
+			expectedProvider:         "openstack",
+			expectedSubscriptionName: "sb-openstack_eu-de-1",
+		},
+		"sap converged cloud eu-de-2": {
+			planID:         broker.SapConvergedCloudPlanID,
+			region:         "eu-de-2",
+			platformRegion: "cf-eu20-staging",
+
+			expectedProvider:         "openstack",
+			expectedSubscriptionName: "sb-openstack_eu-de-2",
+		},
+	} {
+		t.Run(tn, func(t *testing.T) {
+			// given
+			cfg := fixConfig()
+			cfg.Broker.EnablePlans = append(cfg.Broker.EnablePlans, "azure_lite")
```

Comment:
we can add azure_lite to the bsase config (fixConfig), add build-runtime plans to hap-rules file and maybe change the feature flag to use new step in all tests in the package?

Reply:
or just one test for multiple plans run twice (the original TestProvisioning_ClusterParameters)

Reply:
We agreed to have the test just for the new step until we make it a default step in the provisioning queue. We will remove the test after we remove old ResolveCredentials step.